src/commons/BrowserUtils.ts

```diff
@@ -338,50 +354,50 @@ export namespace BrowserUtils {
    *
    * @param handle a string representing a window handle, should be one of the strings that was returned in a call to getWindowHandles
    */
-  export function switchToWindow(handle: string): void {
-    Reporter.debug(`Switching window by id: '${handle}'`);
+  export async function switchToWindow(handle: string): Promise<void> {
+    await Reporter.debug(`Switching window by id: '${handle}'`);
 
-    tryBlock(() => browser.switchToWindow(handle), `Failed switch to window by id: '${handle}'`);
+    await tryBlock(async () => await browser.switchToWindow(handle), `Failed switch to window by id: '${handle}'`);
   }
 
   /**
    * The Get Window Handles command returns a list of window handles for every open top-level browsing context.
    * The order in which the window handles are returned is arbitrary.
    */
-  export function getWindowHandles(): Array<string> {
-    Reporter.debug('Get all ids of all open tabs');
+  export async function getWindowHandles(): Promise<Array<string>> {
+    await Reporter.debug('Get all ids of all open tabs');
 
-    return tryBlock(() => browser.getWindowHandles(), 'Failed to get window handles');
+    return await tryBlock(async () => await browser.getWindowHandles(), 'Failed to get window handles');
```

Comment:
according to `https://webdriver.io/docs/api/jsonwp/#getwindowhandles` this function is sync